### PR TITLE
Refactor Item class in resources.py

### PR DESCRIPTION
**Pull request by sameshl**
related to #424

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.5.2 and above.

### Description

Refactored `if` conditon in resources.py.
Let me know if any thing else needs to be done.
Meanwhile, I am working on adding appropriate test cases for the changes made.

### CODE REVIEW

**Comment by xadahiya** [comment]
Build is failing

**Comment by sameshl** [comment]
@xadahiya Fixing it

**Comment by sameshl** [comment]
@xadahiya I have fixed most of the issues. But it still looks like thier are some build errors. I don't think that is due to my changes. Could you have a look?

**Comment by xadahiya** [comment]
See the log details here https://travis-ci.com/HTTP-APIs/hydrus/jobs/276840266

**Comment by sameshl** [comment]
@xadahiya  @Mec-iS All build errors have been fixed. Please have a look.

**Comment by xadahiya** [comment]
@sameshl Good work! 👏 
However, what is the benefit of doing this? It just adds a new file and several imports, it's not like we're using this particular code several times. Don't you think?

**Comment by sameshl** [comment]
> changes are limited but it is a good example of what we need to do. factorizing is important for readability and later performance improving, extending this practice to more used functions would be nice.

Yes. I am working on factorising other parts of code. I would then proceed to make these write tests for these factored functions improving testing and make functions purpose more consise.

**Comment by sameshl** [comment]
@Mec-iS I refactored the `post`, `put` and `delete` method in the `Item` class. Let me know what else needs to be done.

**Comment by sameshl** [comment]
@Mec-iS what is the status on this? Do I need to correct somethings so reviewers have a look at this?

**Comment by Mec-iS** [comment]
it is up to @xadahiya and @chrizandr to acceppt PR

**Comment by xadahiya** [comment]
@sameshl please remove thr extra imports from resources.py (things that are now imported in itemhelpers py and are redundant). Rest everything looks fine.

**Comment by xadahiya** [comment]
NVM, most of the imports are still being used. Merging

File: hydrus/itemhelpers.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    get_link_props)
+from hydrus.utils import (
+    get_session,
+    get_api_name,
+    get_hydrus_server_url)
+from hydrus.socketio_factory import socketio
+
+
+def items_get_check_support(id_, class_type, class_path, path):
+    """Check if class_type supports GET operation"""
```
Comment (vddesai1871) [inline hydrus/itemhelpers.py]: @xadahiya Names and docstrings of these functions are misleading, we already have a method(named `checkClassOp`) to check whether a class supports a particular method or not.
What we are actually doing here is implementing these methods.

**Pull request merged**